### PR TITLE
Lua/C++ api: Allow to specify what blocks to ignore during raycast

### PR DESCRIPTION
**Pull request by REDxEYE**
libblock.raycast: Add argument to specify blocks to ignore during ray casting

libentity.raycast: Add argument to specify blocks to ignore during ray casting
Chunk::rayCast: Add argument to specify blocks to ignore during ray casting.

On lua side filter blocks are passed as list of strings in form of "MOD:BLOCK_NAME" On C++ size filter blocks are std::set of blockid_t

### CODE REVIEW

**Comment by REDxEYE** [comment]
[trace.zip](https://github.com/user-attachments/files/16664099/trace.zip)
Content pack to test new functionality.
Use key G and H to run block/entity trace

File: doc/en/scripting/builtins/libblock.md [hunk 1]
```diff
@@ -120,12 +120,17 @@ block.raycast(start: vec3, dir: vec3, max_distance: number, [optional] dest: tab
     endpoint: vec3, -- point of the ray hit point
     iendpoint: vec3, -- position of the block hit by the ray
     length: number, -- ray length
```
Comment (MihailRis) [inline doc/en/scripting/builtins/libblock.md:122]: это не список аргументов, а список полей возвращаемой таблицы

File: doc/ru/scripting/builtins/libblock.md [hunk 2]
```diff
@@ -65,12 +65,17 @@ block.raycast(start: vec3, dir: vec3, max_distance: number, [опциональ
     endpoint: vec3, -- точка касания луча
     iendpoint: vec3, -- позиция блока, которого касается луч
     length: number, -- длина луча
```
Comment (MihailRis) [inline doc/ru/scripting/builtins/libblock.md:67]: здесь тоже

File: doc/en/scripting/builtins/libblock.md (hunk 1 above)
Comment (REDxEYE) [reply doc/en/scripting/builtins/libblock.md:122]: Упс, не туда дописал :)

**Pull request merged**